test/test_unistd.rs

```diff
@@ -400,3 +403,55 @@ fn test_pipe2() {
     let f1 = FdFlag::from_bits_truncate(fcntl(fd1, FcntlArg::F_GETFD).unwrap());
     assert!(f1.contains(FdFlag::FD_CLOEXEC));
 }
+
+// Used in `test_alarm`.
+static mut ALARM_CALLED: bool = false;
+
+// Used in `test_alarm`.
+#[no_mangle]
+#[allow(private_no_mangle_fns)]
+pub extern fn alarm_signal_handler(raw_signal: libc::c_int) {
+    assert_eq!(raw_signal, libc::SIGALRM, "unexpected signal: {}", raw_signal);
+    unsafe { ALARM_CALLED = true };
+}
+
+#[test]
+fn test_alarm() {
+    #[allow(unused_variables)]
+    let m = ::SIGNAL_MTX.lock().expect("Mutex got poisoned by another test");
+
+    let handler = SigHandler::Handler(alarm_signal_handler);
+    let signal_action = SigAction::new(handler, SaFlags::SA_RESTART, SigSet::empty());
+    let old_handler = unsafe {
+        sigaction(Signal::SIGALRM, &signal_action)
+            .expect("unable to set signal handler for alarm")
+    };
+
+    // Set an alarm.
+    assert_eq!(alarm(Some(Duration::from_secs(60))), None);
+
+    // Overwriting an alarm should return the old alarm.
+    assert_eq!(alarm(Some(Duration::from_secs(1))), Some(Duration::from_secs(60)));
+
+    // We should be woken up after 1 second by the alarm, so we'll sleep for 2
+    // seconds to be sure.
+    sleep(2);
+    assert_eq!(unsafe { ALARM_CALLED }, true, "expected our alarm signal handler to be called");
+
+    // Reset the signal.
+    unsafe {
+        sigaction(Signal::SIGALRM, &old_handler)
+            .expect("unable to set signal handler for alarm");
+    }
+}
+
+#[test]
+fn test_canceling_alarm() {
+    #[allow(unused_variables)]
```

Comment:
I believe you can eliminate this directive if you change the variable's name from `m` to `_m`.  That didn't work with older Rust releases, but it should be possible now that the minimum Rust is 1.20.0

Reply:
Shall I do it for all uses while I'm at it?

Reply:
If you do, make it a separate commit.

Reply:
Then I'll make a seperate PR for that, including this line.